src/client.c

```diff
@@ -339,13 +343,12 @@ bool set_pool(PgSocket *client, const char *dbname, const char *username, const
 		}
 	} else {
 		client->login_user = find_user(username);
-		if (!client->login_user) {
+		if (!client->login_user || user_requires_auth_query(client->login_user)) {
```

Comment:
Existence of `login_user` does not imply authentication of the user, as the user could exist due to being parsed from `[users]` section, without a password in `auth_file`. This motivates the use of method `user_requires_auth_query`.